npm/generate_katas_content.js

```diff
@@ -566,6 +566,42 @@ function generateKataContent(path, globalCodeSources) {
   return kata;
 }
 
+function validateIdsUniqueness(katas) {
+  console.log("Validating IDs uniqueness across all katas");
+  const allIds = new Set();
+  const assertUniqueness = (id) => {
+    const isIdUnique = !allIds.has(id);
+    if (!isIdUnique) {
+      throw new Error(`"${id}" is not unique`);
+    }
+    allIds.add(id);
+  };
+
+  for (const kata of katas) {
+    // Check kata IDs are unique.
+    assertUniqueness(kata.id);
+    for (const section of kata.sections) {
+      // Check section IDs are unique.
+      assertUniqueness(section.id);
+      if (section.type === "exercise") {
```

Comment:
If we change Kata structure and forget to update this code, we will run into a risk of duplicate ids. But I guess this is OK.

Reply:
I'm not too worried about this because if the structure of katas changes, this code will probably fail anyway and would have to be updated.